test/parallel_api/algorithm/alg.sorting/sort_by_key_common.h

```diff
@@ -0,0 +1,250 @@
+    }
+    else
+    {
+        // Pass a custom comparator to check the corresponding implementation; merge-sort may be used
+        auto greater = [](const auto& lhs, const auto& rhs) { return lhs > rhs; };
+        if constexpr (std::is_same_v<StabilityTag, StableSort>)
+            oneapi::dpl::stable_sort_by_key(policy, keys_begin, keys_begin + n, vals_begin, greater);
+        else
+            oneapi::dpl::sort_by_key(policy, keys_begin, keys_begin + n, vals_begin, greater);
+    }
+}
+
+template<typename _KeysIt, typename _ValsIt, typename _Size, typename StabilityTag, typename DirectionTag>
+void
+check_sort(const _KeysIt& keys_begin, const _ValsIt& vals_begin, _Size n, StabilityTag, DirectionTag)
+{
+    if constexpr (std::is_same_v<DirectionTag, AscendingSort>)
+    {
+        const int k = (n - 1) / 2 + 1;
+        for (int i = 0; i < n; ++i)
+        {
+            if(i - k < 0)
+            {
+                //a key should be 1 and value should be even
+                EXPECT_TRUE(keys_begin[i] == 1 && vals_begin[i] % 2 == 0, "wrong result with a standard policy");
+            }
+            else
+            {
+                //a key should be 2 and value should be odd
+                EXPECT_TRUE(keys_begin[i] == 2 && vals_begin[i] % 2 == 1, "wrong result with a standard policy");
+            }
+        }
+
+        if constexpr (std::is_same_v<StabilityTag, StableSort>)
+        {
+            EXPECT_TRUE(std::is_sorted(vals_begin, vals_begin + k),
+                        "wrong result with a standard policy, sort stability issue");
+            EXPECT_TRUE(std::is_sorted(vals_begin + k, vals_begin + n),
+                        "wrong result with a standard policy, sort stability issue");
+        }
+    }
+    else
+    {
+        EXPECT_TRUE(std::is_sorted(keys_begin, keys_begin + n, std::greater<void>()), "wrong result with hetero policy, USM data");
+        if constexpr (std::is_same_v<StabilityTag, StableSort>)
+        {
+            EXPECT_TRUE(std::is_sorted(vals_begin, vals_begin + n),
+                        "wrong result with hetero policy, USM data, sort stability issue");
+        }
+    }
+}
+
+template<typename _Policy, typename StabilityTag, typename DirectionTag>
+void
+test_with_std_policy(_Policy&& policy, StabilityTag stability_tag, DirectionTag direction_tag)
+{
+    constexpr int n = 1000000;
+    std::vector<int> keys_buf(n);
+    std::vector<int> vals_buf(n);
+
+    generate_data(keys_buf.begin(), vals_buf.begin(), n, direction_tag);
+    call_sort(policy, keys_buf.begin(), vals_buf.begin(), n, stability_tag, direction_tag);
+    check_sort(keys_buf.begin(), vals_buf.begin(), n, stability_tag, direction_tag);
+}
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <sycl::usm::alloc alloc_type, typename StabilityTag, typename DirectionTag>
+void
+test_with_usm(sycl::queue& q, StabilityTag stability_tag, DirectionTag direction_tag)
+{
+    constexpr int n = (1 << 12) + 42;
```

Comment:
Such magic formulas are difficult to understand....
I see that (1 << 12) :  12th degree of 2.... but why 42? 
Overall, such formulas should be explained in a comment.

Reply:
I think its just a number that is non-zero so we aren't just testing powers of 2 or powers of 10.

42 is commonly used because of Hitchhikers guide:
[The Answer to the Ultimate Question of Life, the Universe, and Everything is 42](https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy)

Reply:
I've replaced this formula with a constant. You can find it as `small_size` variable, which accompanies `large_size`. They contain some "random" values, which are non-zero, non-power of 2 or 10 as described by Dan. Should it still be explained in a comment?

Reply:
I've additionally explained it.